pkg/policyengine/connectivitypdp/connectivity_pdp_test.go

```diff
@@ -241,9 +254,20 @@ func addPoliciesFromFile(pdp *connectivitypdp.PDP, filename string) error {
 			return err
 		}
 
-		err = pdp.AddOrUpdatePolicy(&policy)
-		if err != nil {
-			fmt.Printf("invalid connectivity policy: %v\n", err)
+		if policy.Kind == "PrivilegedAccessPolicy" {
```

Comment:
Is the use of the intermediate representation necessary?
I'm not fully clear how much is point in time and how much is final code, post split to different kinds? 😕

Reply:
Testing will be improved (as the PR description says). I should probably avoid file-based tests and switch to struct-based